examples/multi-agent-debate/main.py

```diff
@@ -0,0 +1,239 @@
+import typer
+import logging
+
+from rich.prompt import Prompt
+from typing import List, Any
+
+import langroid as lr
+from langroid.language_models import OpenAIGPTConfig
+from langroid.agent.chat_agent import ChatAgent, ChatAgentConfig
+from langroid.agent.task import Task
+from langroid.agent.tools.metaphor_search_tool import MetaphorSearchTool
+from langroid.utils.logging import setup_logger
+from langroid.agent.tools.orchestration import DoneTool
+from langroid.agent.tools.orchestration import AgentDoneTool
+from langroid import ChatDocument, Entity
+
+from config import get_base_llm_config, get_global_settings
+from models import SystemMessages, load_system_messages
+from system_messages import (
+    DEFAULT_SYSTEM_MESSAGE_ADDITION,
+    FEEDBACK_AGENT_SYSTEM_MESSAGE,
+    generate_metaphor_search_agent_system_message,
+)
+# Import from utils.py
+from utils import (
+    select_model,
+    select_topic_and_setup_side,
+    is_llm_delegate,
+    is_metaphor_search_key_set,
+    is_same_llm_for_all_agents,
+    select_max_debate_turns,
+)
+
+
+class MetaphorSearchChatAgent(ChatAgent):
+    def handle_message_fallback(
+            self, msg: str | ChatDocument
+    ) -> AgentDoneTool | None:
+        """Handle scenario where LLM did not generate any Tool"""
+        if isinstance(msg, ChatDocument) and msg.metadata.sender == Entity.LLM:
+            return AgentDoneTool(content="done")
```

Comment:
When I ran with gpt4o it again got refs but did not argue the 2 sides. 
One way to handle that is to change this line to return a message:

```python
return f"Have you presented pro and con arguments based on your search results? If so, use the TOOL `{DoneTool.name()}` to indicate you're finished. Otherwise, argue both sides and then send the `{DoneTool.name()}`
```

Reply:
the LLM will see this "nudge" and respond accordingly

Reply:
I changed the return message and tested the output. Screenshot attached 
![DoneTool_return_message](https://github.com/user-attachments/assets/d63c9193-fe26-43dc-9aee-0efa241a4891)